### PR TITLE
fix: Delete duplicate ticket tags

### LINKED ISSUE #6382: Ticket tag migrations failing for production data
```
sqlalchemy.exc.IntegrityError: (psycopg2.errors.UniqueViolation) could not create unique index "unique_ticket_tag"
DETAIL:  Key (name, event_id)=(free, 498) is duplicated.

[SQL: ALTER TABLE ticket_tag ADD CONSTRAINT unique_ticket_tag UNIQUE (name, event_id)]
```

### PR DESCRIPTION
**Pull request by prateekj117**
Fixes #6382 

#### Checklist

- [X] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [X] My branch is up-to-date with the Upstream `development` branch.
- [ ] The unit tests pass locally with my changes 
- [ ] I have added tests that prove my fix is effective or that my feature works
- [X] I have added necessary documentation (if appropriate)

- [X] All the functions created/modified in this PR contain relevant docstrings.

### CODE REVIEW

**Comment by prateekj117** [comment]
@iamareebjamal I changed the revision and downrevision appropriately. And I am also appending '.deleted' to the ticket-tag.

File: migrations/versions/rev-2019-08-13-14:38:26-1bdf23be63bd_.py [hunk 1]
```diff
@@ -12,8 +12,8 @@
 
 
 # revision identifiers, used by Alembic.
-revision = '1bdf23be63bd'
-down_revision = '246b5b6123af'
+revision = '9172ceb350b8'
```
Comment (iamareebjamal) [inline migrations/versions/rev-2019-08-13-14:38:26-1bdf23be63bd_.py]: Revision should remain unchanged

**Comment by prateekj117** [comment]
@iamareebjamal Done.

**Comment by iamareebjamal** [comment]
This will work, but what about tickets linked with the deleted ticket tag?

**Comment by prateekj117** [comment]
@iamareebjamal I couldn't find where it's saving the information about the ticket linked with the ticket-tag.

**Comment by iamareebjamal** [comment]
Ran your migration. There are still duplicates and the proceeding migration failed. What did you even test?

@CosmicCoder96 Are ticket tags even being used, there is no entry in association table?

**Comment by prateekj117** [comment]
@iamareebjamal I tested it using `in` and not `not in` because I had the unique constraint already. 
Sorry for the Negligence.
Weirdly enough, it turns out `not in` is returning null values.

**Comment by iamareebjamal** [comment]
Then it is wrong query

**Comment by prateekj117** [comment]
@iamareebjamal I have corrected the query. Please review.

**Comment by iamareebjamal** [comment]
Doesn't work if there are more than 2 duplicates

**Comment by CosmicCoder96** [comment]
> @CosmicCoder96 Are ticket tags even being used, there is no entry in association table?

@iamareebjamal I don't think they are consumed by current FE, but they are anyway supported by the ticketing API, referred to multiple times in it. And the only ticket tags I see are for some events on legacy data.

**Comment by iamareebjamal** [comment]
@prateekj117 Wrap this up ASAP, please

**Comment by prateekj117** [comment]
@iamareebjamal It's working even when there are more than two duplicates:

Screenshot from 2019-08-18 22-05-56

**Comment by iamareebjamal** [comment]
More than 2 duplicates of 1 ticket tag!

**Comment by iamareebjamal** [comment]
Yeah, so exactly @prateekj117 

Do you think the unique constraint will work on the data above?

**Comment by iamareebjamal** [comment]
(15, paid !!!!.deleted) is still duplicated. So how it is going to work?

**Comment by prateekj117** [comment]
@iamareebjamal Ok, I got the point.

**Comment by prateekj117** [comment]
@iamareebjamal Should I delete the rows?

**Comment by iamareebjamal** [comment]
No

**Comment by prateekj117** [comment]
@iamareebjamal Can you give me a rough idea on how to go about for a query like that?

**Comment by iamareebjamal** [comment]
Append id or fixed random characters before .deleted

And change the implementation of downgrade accordingly

**Comment by prateekj117** [comment]
@iamareebjamal Done.

**Comment by iamareebjamal** [comment]
@CosmicCoder96 Please verify

**Pull request merged**